easybuild/easyblocks/p/python.py

```diff
@@ -128,6 +132,39 @@ def __init__(self, *args, **kwargs):
             easybuild_subdir = log_path()
             self.pythonpath = os.path.join(easybuild_subdir, 'python')
 
+        ext_defaults = {
+            # The only installed packages at this point are default installed ones, e.g. pip&setuptools
+            # And we want to upgrade them cleanly, i.e. uninstall them
+            'pip_ignore_installed': False,
+            # Python installations must be clean. Requires pip >= 9
+            'sanity_pip_check': LooseVersion(self._get_pip_ext_version() or '0') >= LooseVersion('9'),
```

Comment:
This comment should be clarified a bit I think, and I would use `9.0` rather than `9` (just for clarity sake):

```suggestion
            # enable "pip check" in sanity check by default, to ensure that Python installations are clean;
            # requires pip v9.0 or newer
            'sanity_pip_check': LooseVersion(self._get_pip_ext_version() or '0') >= LooseVersion('9.0'),
```

That said, I don't think we should be enabling `sanity_pip_check` by default just yet. It'll result in broken `Python` easyconfigs that worked just fine up until now, and changes that break backwards compability should only be made in the next major EasyBuild release (v5.0), cfr. https://github.com/easybuilders/easybuild-easyblocks/issues/2127

Reply:
Agreed. Removed and will open a follow-up PR for this

Reply:
Follow-up: https://github.com/easybuilders/easybuild-easyblocks/pull/2423